src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -596,7 +596,7 @@ private function createUserProviders($config, ContainerBuilder $container)
         $providerIds = array();
         foreach ($config['providers'] as $name => $provider) {
             $id = $this->createUserDaoProvider($name, $provider, $container);
-            $providerIds[] = $id;
+            $providerIds[$name] = $id;
```

Comment:
or normalize here :)

Reply:
good catch, not sure about handling duplicates now